application/frontend/controllers/UserController.php

```diff
@@ -4,17 +4,49 @@
 use common\models\User;
 use frontend\components\BaseRestController;
 use Yii;
-use yii\web\NotFoundHttpException;
 
 class UserController extends BaseRestController
 {
+    public function actionIndex() // GET /user
+    {
+        return User::find()->all();
+    }
+
+    public function actionView($employeeId) // GET /user/abc123
+    {
+        $user = User::findOne(['employee_id' => $employeeId]);
+
+        if (isset($user)) {
+            return $user;
+        }
+
+        Yii::$app->response->statusCode = 204;
+    }
+
     public function actionCreate(): User
     {
-        $existingUser = User::findOne([
-            'employee_id' => Yii::$app->request->getBodyParam('employee_id')
-        ]);
+        $user = new User();
+
+        $user->scenario = User::SCENARIO_NEW_USER;
+
+        $user->attributes = Yii::$app->request->getBodyParams();
+
+        parent::save($user);
```

Comment:
Hmm... you may be right about `parent::someMethod()` not necessarily implying static. The only places I'm using to seeing `parent::` are when calling the parent version of the current method (such as calling `parent::rules()` within the current class's `rules()` method). When calling a non-static method the common practice seems to be to use `$this` regardless of whether the method is defined in the current class or a parent class.

The lack of a complaint from PHP could simply be related to special details about `parent::` (as opposed to `ClassName::`). I'm not sure on that one.